articles/how-to-upload-files-to-aws-s3-using-django-rest-framework/index.md

```diff
@@ -0,0 +1,437 @@
+
+MEDIA_URL = '/media/'
+
+MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
+
+```
+
+In **dropboxer/urls.py**:
+
+```python
+
+from django.conf import settings  # new
+
+from django.conf.urls.static import static  # new
+
+from django.contrib import admin
+
+from django.urls import path
+
+urlpatterns = [
+
+    path('admin/', admin.site.urls),
+
+]
+
+if settings.DEBUG:
+
+    urlpatterns += static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
+
+    urlpatterns += static(settings.STATIC_URL, document_root=settings.STATIC_ROOT)
+
+```
+
+Create ***serializers.py*** and ***urls.py*** files in the ***uploader*** app.
+
+In ***models.py***, we create a simple model that uploads files.
+
+```python
+
+from django.db import models
+
+ 
+
+class DropBox(models.Model):
+
+    title = models.CharField(max_length=30)
+
+    document = models.FileField(max_length=30)
+
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    updated_at = models.DateTimeField(auto_now=True)
+
+ 
+
+    class Meta:
+
+        verbose_name_plural = 'Drop Boxes'
+
+```
+
+In ***serializers.py***:
+
+```python
+
+from rest_framework import serializers
+
+from .models import DropBox
+
+class DropBoxSerializer(serializers.ModelSerializer):
+
+ 
+
+    class Meta:
+
+        model = DropBox
+
+        fields = '__all__'
+
+```
+
+In **uploader/views.py**:
+
+```python
+
+from rest_framework import viewsets, parsers
+
+from .models import DropBox
+
+from .serializers import DropBoxSerializer
+
+class DropBoxViewset(viewsets.ModelViewSet):
+
+ 
+
+    queryset = DropBox.objects.all()
+
+    serializer_class = DropBoxSerializer
+
+    parser_classes = [parsers.MultiPartParser, parsers.FormParser]
+
+    http_method_names = ['get', 'post', 'patch', 'delete']
+
+```
+
+In **uploader/urls.py**:
+
+```python
+
+from rest_framework.routers import SimpleRouter
+
+from .views import DropBoxViewset
+
+router = SimpleRouter()
+
+router.register('accounts', DropBoxViewset)
+
+urlpatterns = router.urls
+
+```
+
+In **uploader/admin**:
+
+```python
+
+from django.contrib import admin
+
+from .models import DropBox
+
+admin.site.register(DropBox)
+
+```
+
+In **dropboxer/urls.py**:
+
+```python
+
+...
+
+from django.urls import path, include  # new
+
+urlpatterns = [
+
+    path('admin/', admin.site.urls),
+
+    path('api/', include('rest_framework.urls')),  # new
+
+    path('', include('uploader.urls')),  # new
+
+]
+
+```
+
+Make migrations in terminal:
+
+```bash
+
+python manage.py makemigrations
+
+python manage.py migrate
+
+```
+
+Start the local server in the terminal:
+
+```bash
+
+python manage.py runserver
+
+```
+
+Visit the running server on a [web browser](http://localhost:8000/accounts/). We find the sample file uploaded in the **media** folder. This sample file was created when we uploaded it via the API on the browser as shown below:
+
+![media_folder_included_vs_code](../how-to-upload-files-to-aws-s3-using-django-rest-framework/media_included_structure.png)
```

Comment:
Consider changing all the image paths by appending `/engineering-education/how-to-upload-files-to-aws-s3-using-django-rest-framework/` before every the image name. For example, in this above image change `![media_folder_included_vs_code](../how-to-upload-files-to-aws-s3-using-django-rest-framework/media_included_structure.png)` to `![media_folder_included_vs_code](engineering-education/how-to-upload-files-to-aws-s3-using-django-rest-framework/media_included_structure.png)`